docs/disaster-recovery.md

```diff
@@ -0,0 +1,399 @@
+
+>:memo: **NOTE**: rbd-mirror assumes the two clusters are named differently.
+
+## Deploy the Volume Replication CRD
+
+Volume Replication Operator follows controller pattern and provides extended APIs for storage disaster recovery. The extended APIs are provided via Custom Resource Definition (CRD).
+
+>:bulb: For more information, please refer to the [volume-replication-operator](https://github.com/csi-addons/volume-replication-operator).
+
+* Deploy the `VolumeReplicationClass` CRD
+
+    ```bash
+    kubectl create -f https://raw.githubusercontent.com/csi-addons/volume-replication-operator/main/config/crd/bases/replication.storage.openshift.io_volumereplicationclasses.yaml
+    ```
+
+    >```bash
+    > customresourcedefinition.apiextensions.k8s.io/volumereplicationclasses.replication.storage.openshift.io created
+    >```
+
+* Deploy the VolumeReplication CRD
+
+    ```bash
+   kubectl create -f https://raw.githubusercontent.com/csi-addons/volume-replication-operator/main/config/crd/bases/replication.storage.openshift.io_volumereplications.yaml
+    ```
+
+    >```bash
+    > customresourcedefinition.apiextensions.k8s.io/volumereplications.replication.storage.openshift.io created created
+    >```
+
+### Add RBAC rules for Volume Replication Operator
+
+Add the below mentioned rules to `rbd-external-provisioner-runner` ClusterRole in [csi-provisioner-rbac.yaml](https://github.com/ceph/ceph-csi/blob/devel/deploy/rbd/kubernetes/csi-provisioner-rbac.yaml)
+
+```yaml=61
+  - apiGroups: ["replication.storage.openshift.io"]
+    resources: ["volumereplications", "volumereplicationclasses"]
+    verbs: ["create", "delete", "get", "list", "patch", "update", "watch"]
+  - apiGroups: ["replication.storage.openshift.io"]
+    resources: ["volumereplications/finalizers"]
+    verbs: ["update"]
+  - apiGroups: ["replication.storage.openshift.io"]
+    resources: ["volumereplications/status"]
+    verbs: ["get", "patch", "update"]
+  - apiGroups: ["replication.storage.openshift.io"]
+    resources: ["volumereplicationclasses/status"]
+    verbs: ["get"]
+```
+
+### Deploy the Volume Replication Sidecar
+
+Volume Replication Operator is a kubernetes operator that provides common and reusable APIs for storage disaster recovery. It is based on [csi-addons/spec](https://github.com/csi-addons/spec) specification and can be used by any storage provider.
+
+To deploy `volume-replication` sidecar container in `csi-rbdplugin-provisioner` pod, add the following yaml to [csi-rbdplugin-provisioner deployment](https://github.com/ceph/ceph-csi/blob/devel/deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml).
+
+```yaml=1
+        - name: volume-replication
+          image: quay.io/csiaddons/volumereplication-operator:v0.1.0
+          args :
+            - "--metrics-bind-address=0"
+            - "--leader-election-namespace={{ .Namespace }}"
+            - "--driver-name={{ .DriverNamePrefix }}rbd.csi.ceph.com"
```

Comment:
`DriverNamePrefix` is not required and namespace need to  be replaced by env variable refer to existing deployment.